src/provider/systemdmanager/service.rs

```diff
@@ -0,0 +1,321 @@
+use std::collections::HashMap;
+
+use kubelet::container::Container;
+use kubelet::pod::Pod;
+use phf::{phf_map, phf_ordered_set};
+
+use crate::provider::error::StackableError;
+
+use crate::provider::error::StackableError::PodValidationError;
+use crate::provider::states::creating_config::CreatingConfig;
+use crate::provider::PodState;
+use log::{debug, error, trace, warn};
+
+static RESTART_POLICY_MAP: phf::Map<&'static str, &'static str> = phf_map! {
+    "Always" => "always",
+    "OnFailure" => "on-failure",
+    "Never" => "no",
+};
+
+// TODO: This will be used later to ensure the same ordering of known sections in
+// unit files, I'll leave it in for now
+#[allow(dead_code)]
+static SECTION_ORDER: phf::OrderedSet<&'static str> =
```

Comment:
It can in theory also be a Vec, the benefit of this is the compile time check against duplicate entries, which Vec wouldn't give.
Granted, very small chance of that happening..